packages/cipherstash-proxy-integration/src/multitenant/set_keyset_id.rs

```diff
@@ -246,6 +246,44 @@ mod tests {
         assert!(result.is_err());
     }
 
+    ///
+    /// Tests error handling of unknown keyset id
+    ///
+    #[tokio::test]
+    async fn set_keyset_id_unknown() {
+        trace();
+
+        clear().await;
+
+        let client = connect_with_tls(PROXY).await;
+
+        let tenant_keyset_id_1 = std::env::var("CS_TENANT_KEYSET_ID_1")
+            .map(|s| Uuid::parse_str(&s).unwrap())
+            .unwrap();
+        // Can set unknown name
+        let sql = "SET CIPHERSTASH.KEYSET_ID = '2cace9db-3a2a-4b46-a184-ba412b3e0730'";
+        let result = client.query(sql, &[]).await;
+        assert!(result.is_ok());
+
+        // Error on encrypt
+        let id = random_id();
+        let text = "TEST UNKNOWN";
+
+        let insert_sql = "INSERT INTO encrypted (id, encrypted_text) VALUES ($1, $2)";
+        let result = client.query(insert_sql, &[&id, &text]).await;
+        assert!(result.is_err());
```

Comment:
I think this is fine, but just out of curiosity, is it possible/worthwhile to check the error types?

Reply:
I was thinking the same thing after I pushed this, actually. 
And probably worth extending across the other error tests.